### PR TITLE
[FIX] add namespace seqan3::align_cfg::detail

### PR DESCRIPTION
**Pull request by marehr**
Blocks #2031 and #2020

### CODE REVIEW

File: include/seqan3/alignment/configuration/all.hpp [hunk 1]
```diff
@@ -31,6 +31,12 @@
  * \brief A special sub namespace for the alignment configurations.
  */
 
+ /*!\cond DEV
+  * \namespace seqan3::align_cfg::detail
+  * \copydoc seqan3::detail
+  * \endcond
+  */
```
Comment (eseiler) [inline include/seqan3/alignment/configuration/all.hpp:38]:
```suggestion
 /*!\if DEV
  * \namespace seqan3::align_cfg::detail
  * \copydoc seqan3::detail
  * \endif
  */
```

We should use `if` for inside comment blocks and `cond` for exluding code / multiple comment blocks

Comment (marehr) [reply include/seqan3/alignment/configuration/all.hpp:38]: Is this the same?

I just copied the code from

https://github.com/seqan/seqan3/blob/aa45291d7aafd88d2d58c2e4fcf1702b1fbf6220/include/seqan3/core/all.hpp#L48-L56

Comment (eseiler) [reply include/seqan3/alignment/configuration/all.hpp:38]: if would be better style; but then you can just leave it as it is.

Comment (marehr) [reply include/seqan3/alignment/configuration/all.hpp]: Changed it

**Pull request merged**